xalan/src/main/java/org/apache/xalan/Version.java

```diff
@@ -20,36 +20,99 @@
  */
 package org.apache.xalan;
 
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * Administrative class to keep track of the version number of
  * the Xalan release.
  * <P>This class implements the upcoming standard of having
- * org.apache.project-name.Version.getVersion() be a standard way 
- * to get version information.  This class will replace the older 
+ * org.apache.project-name.Version.getVersion() be a standard way
+ * to get version information.  This class will replace the older
  * org.apache.xalan.processor.Version class.</P>
- * <P>See also: org/apache/xalan/res/XSLTInfo.properties for 
+ * <P>See also: org/apache/xalan/res/XSLTInfo.properties for
  * information about the version of the XSLT spec we support.</P>
  * @xsl.usage general
  */
 public class Version
 {
+  private static final String POM_PROPERTIES_JAR = "org/apache/xalan/version.properties";
+  private static final String POM_PROPERTIES_FILE_SYSTEM = "xalan/target/classes/" + POM_PROPERTIES_JAR;
```

Comment:
Good catch! I added that, because I had tried something from the command line, probably without setting the proper classpath. When running from the IDE or from a hypothetical unit or integration test, the corresponding code branch should never be entered. I am going to remove it.